src/core/runtime/Parallelizer_utils.cpp

```diff
@@ -112,11 +119,21 @@ class DispatcherInfo {
  * Dispatch threads to run a DOALL loop.
  */
 DispatcherInfo NOELLE_DOALLDispatcher(
-    void (*parallelizedLoop)(void *, int64_t, int64_t, int64_t),
+    void (*parallelizedLoop)(void *, int64_t, int64_t, int64_t, void *),
     void *env,
     int64_t maxNumberOfCores,
     int64_t chunkSize);
 
+int NOELLE_paraPrintfOneInt(
+    int64_t coreID,
+    int64_t numCores,
+    int64_t chunkSize,
+    ThreadSafeLockFreeQueue<NOELLE_OutputMessage_t> *outputQueue,

```

Comment:
Are you going to have a single producer and a single consumer for this queue?

Reply:
Yes, one queue per task thread.